cmd/kraft/run/runner_project.go

```diff
@@ -38,14 +38,21 @@ func (runner *runnerProject) String() string {
 func (runner *runnerProject) Runnable(ctx context.Context, opts *Run, args ...string) (bool, error) {
 	var err error
 
+	cwd, err := os.Getwd()
+	if err != nil {
+		return false, err
+	}
+
 	if len(args) == 0 {
-		runner.workdir, err = os.Getwd()
-		if err != nil {
-			return false, err
-		}
+		runner.workdir = cwd
 	} else {
-		runner.workdir = args[0]
-		runner.args = args[1:]
+		if f, err := os.Stat(args[0]); err == nil && f.IsDir() {
+			runner.workdir = args[0]
+			runner.args = args[1:]
+		} else {
+			runner.workdir = cwd
+			runner.args = args
+		}
```

Comment:
Style suggestion to avoid the `else` (feel free to discard)
```suggestion
		runner.workdir = cwd
		runner.args = args
		if f, err := os.Stat(args[0]); err == nil && f.IsDir() {
			runner.workdir = args[0]
			runner.args = args[1:]
		}
```